src/Generator/EnumGenerator/EnumGenerator.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Laminas\Code\Generator\EnumGenerator;
+
+use Laminas\Code\Generator\EnumGenerator\Cases\CaseFactory;
+use Laminas\Code\Generator\MethodGenerator;
+
+use function array_key_exists;
+use function array_map;
+use function implode;
+
+final class EnumGenerator
```

Comment:
This could potentially have a `/** @psalm-immutable */` attached

Reply:
Unfortunately we can't mark it as immutable because it's depending on the `MethodGenerator` and that class is not immutable :'(